numpy/core/src/npymath/halffloat.c

```diff
@@ -23,93 +23,68 @@
 float npy_half_to_float(npy_half h)
 {
     union { float ret; npy_uint32 retbits; } conv;
-    conv.retbits = npy_halfbits_to_floatbits(h);
+    conv.retbits = npy_halfbits_to_floatbits(h.value);
     return conv.ret;
 }
 
 double npy_half_to_double(npy_half h)
 {
     union { double ret; npy_uint64 retbits; } conv;
-    conv.retbits = npy_halfbits_to_doublebits(h);
+    conv.retbits = npy_halfbits_to_doublebits(h.value);
     return conv.ret;
 }
 
 npy_half npy_float_to_half(float f)
 {
     union { float f; npy_uint32 fbits; } conv;
     conv.f = f;
-    return npy_floatbits_to_halfbits(conv.fbits);
+    return (npy_half){npy_floatbits_to_halfbits(conv.fbits)};
 }
 
 npy_half npy_double_to_half(double d)
 {
     union { double d; npy_uint64 dbits; } conv;
     conv.d = d;
-    return npy_doublebits_to_halfbits(conv.dbits);
-}
-
-int npy_half_iszero(npy_half h)
-{
-    return (h&0x7fff) == 0;
-}
-
-int npy_half_isnan(npy_half h)
-{
-    return ((h&0x7c00u) == 0x7c00u) && ((h&0x03ffu) != 0x0000u);
-}
-
-int npy_half_isinf(npy_half h)
-{
-    return ((h&0x7fffu) == 0x7c00u);
-}
-
-int npy_half_isfinite(npy_half h)
```

Comment:
Hmm, I think we have to keep the stubs here to ensure that the symboll remains included in the `npymath` library that we export?  I.e. someone who compiled against a NumPy without the inline definition in the header will need to load these functions from the library dynamiclaly.
As far as I understand, keeping an `int npy_half_isfinite()` around here will ensure that they are still inlucded?